### ISSUE TITLE
gridOptions.columnApi.getColumnGroup() should return the column group regardless of its visibility.

### ISSUE THREAD

**Issue by timmehmainframe** [ceolter/ag-grid#696]
I'm trying to toggle the visibility of columns by their group.
```
var groupColumn = $scope.gridOptions.columnApi.getColumnGroup('myGroupId');
$scope.gridOptions.columnApi.setColumnsVisible(groupColumn.children, true);
```

Example: http://jsbin.com/cihapifema/edit?js,console,output

gridOptions.columnApi.getColumn() returns a column regardless if its visible, so getColumnGroup() should do the same?

**Comment by ceolter** [comment]
i understand why you think this, and it's a fair assumption, but it's just not how things work! let me try and explain:

the columns always exist, exactly one column for every column def. the column then has a 'visible' attribute.

the groups are transient and only exists if they are needed (there is also the concept of 'OriginalColumnGroup' to keep track of what columns are in each group, but you don't have access to that, it's internal)

the groups, in addition to transient, can be, what's the best word, cloned??? suppose you have 10 cols in group A and 10 cols in group B, that's two groups. but then if you reorder the groups, so it's every second one, you end up with 20 displayed column groups, each group having just one column, because you arrange the columns in such a way that they get split. the last example on this page shows it:
https://www.ag-grid.com/angular-grid-master-slave/index.php

so, in summary, column groups only exists if the group is showing, and there can be multiple groups for the same group if the columns are split.

so, that's why the column groups don't return if they are not visible!
